src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -729,153 +728,291 @@ func (s *service) FetchTagged(tctx thrift.Context, req *rpc.FetchTaggedRequest)
 		)
 	}
 
-	result, err := s.fetchTagged(ctx, db, req)
+	result, err := s.fetchTagged(ctx, req)
 	if sampled && err != nil {
 		sp.LogFields(opentracinglog.Error(err))
 	}
 	sp.Finish()
 
+	if err != nil {
+		s.metrics.fetchTagged.ReportError(s.nowFn().Sub(callStart))
+	} else {
+		s.metrics.fetchTagged.ReportSuccess(s.nowFn().Sub(callStart))
+	}
```

Comment:
Can use `.ReportSuccessOrError(err, s.nowFn().Sub(callStart))` here and it will do the nil check of the error for you.

Reply:
nice. just had the code from before.